### PR TITLE
Feature/admin

### PR DESCRIPTION
**Pull request by zuccha**

### CODE REVIEW

File: test/routes-test/admin-delete.js [hunk 1]
```diff
@@ -100,6 +173,86 @@ describe('Admin-delete test: ', function() {
...
+            done(err);
+          } else {
+            res.body.should.have.property("error", "user is not claiming");
+            done();
+          }
+        });
+    });
+
+    // WRONG
+    it('should respond with 400 if user is already associtated with a freelancer', function(done) {
```
Comment (lazy-val) [inline test/routes-test/admin-delete.js]: Typo (also on the next test) ;P

**Review by lazy-val — APPROVED** [review]
Very nice overall!

The status codes 451, 452 and 453 are a bit weird, but in the end it doesn't really matter.

For the error messages on the admin login you could use the "element.setCustomValidity" function that Aron and me also used for the username on the registration. However, it's not important for this sprint.

File: public/components/admin.js [hunk 2]
```diff
@@ -3,6 +3,20 @@
 let g_username;
```
Comment (Spyridox) [inline public/components/admin.js]: Why the `g`?

Comment (aronfiechter) [inline public/components/admin.js:3]: Why the `g` (also below)?

**Review by aronfiechter — APPROVED** [review]
I skim read it, but I tested quite a lot using the system (also trying to claim, reject, then claim again, tested the emails by uncommenting the code).

File: public/components/admin.js (hunk 2 above)
Comment (zuccha) [reply public/components/admin.js:3]: Global. I can remove it, but I think it's better to keep it...

File: test/routes-test/admin-delete.js (hunk 1 above)
Comment (zuccha) [reply test/routes-test/admin-delete.js]: Fixed.

File: public/components/admin.js (hunk 2 above)
Comment (larabr) [reply public/components/admin.js:3]: I'd just argue that it should be `gUsername` but ok

**Pull request merged**